src/btree/bt_curnext.c

```diff
@@ -16,9 +16,10 @@ static inline int
 __cursor_fix_append_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart)
 {
     WT_SESSION_IMPL *session;
-    WT_UPDATE *upd;
+    WT_UPDATE_VIEW upd_view;
```

Comment:
Instead of allocating/clearing this on each call, did you consider adding it to the `WT_CURSOR_BTREE` structure, and re-using it? Additionally, reusing the internal `WT_ITEM` memory as well.

Reply:
Yeah, perhaps this is a good idea. It sounds like a nice middle ground between what I have now and passing the cursor's value buffer down the stack and using it directly.

Reply:
I also prefer it in the cursor.

Reply:
c002170